media_driver/linux/common/ddi/media_libva_util.cpp

```diff
@@ -405,6 +411,23 @@ VAStatus DdiMediaUtil_AllocateSurface(
     uint32_t    gmmPitch;
     uint32_t    gmmSize;
     uint32_t    gmmHeight;
+
+    gmmPitch    = (uint32_t)gmmResourceInfo->GetRenderPitch();
+    if( DdiMediaUtil_IsExternalSurface(mediaSurface) && ( mediaSurface->pSurfDesc->uiVaMemType == VA_SURFACE_ATTRIB_MEM_TYPE_USER_PTR ) )
+    {
+        uint32_t gmmHAlign;
+        gmmHAlign = gmmResourceInfo->GetHAlign();
+        if((gmmPitch < mediaSurface->pSurfDesc->uiPitches[0]) && (((mediaSurface->pSurfDesc->uiPitches[0]-gmmPitch)%gmmHAlign) == 0))
+        {
+            gmmResourceInfo->OverridePitch(mediaSurface->pSurfDesc->uiPitches[0]);
```

Comment:
I still don't understand why we need to call OverridePitch? If we notified gmmlib that memory is user allocated, from my perspective gmmlib should return user allocated pitch right away... So, I am not sure that this entire code block is actually needed.

Reply:
1. there are no interface to set the pitch from application except the "OverridePitch"

Reply:
In this case, how CreateResInfoObject() manages to validate anything? I can't imagine memory allocation validation knowing nothing about the pitch.

Reply:
yes, let me add engineer from https://github.com/intel/gmmlib , @drprajap, @kmvadsar